From cfb80587da7bf3d6a8eeb9b846702d6d731aa1c6 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Wed, 9 Apr 2025 11:32:08 +0200
Subject: [PATCH] libmount: (subdir) remove unused code

The optlist already handles quoted values, so there's no need to do it
in the callers.

Signed-off-by: Karel Zak <kzak@redhat.com>
(cherry picked from commit 5462fa3435544344727b8644205ae427dfd5fcba)
---
 libmount/src/hook_subdir.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/libmount/src/hook_subdir.c b/libmount/src/hook_subdir.c
index 5949af7d8..1e5d79958 100644
--- a/libmount/src/hook_subdir.c
+++ b/libmount/src/hook_subdir.c
@@ -329,9 +329,6 @@ static int is_subdir_required(struct libmnt_context *cxt, int *rc, char **subdir
 
 	dir = mnt_opt_get_value(opt);
 
-	if (dir && *dir == '"')
-		dir++;
-
 	if (!dir || !*dir) {
 		DBG(HOOK, ul_debug("failed to parse X-mount.subdir '%s'", dir));
 		*rc = -MNT_ERR_MOUNTOPT;
-- 
2.49.0

